code/Forms/UsedOnTable.php

```diff
@@ -0,0 +1,158 @@
+ */
+class UsedOnTable extends FormField
+{
+    protected $schemaDataType = FormField::SCHEMA_DATA_TYPE_CUSTOM;
+
+    /**
+     * @skipUpgrade
+     * @var string
+     */
+    protected $schemaComponent = 'UsedOnTable';
+
+    /**
+     * @config
+     * @var array
+     */
+    private static $allowed_actions = [
+        'usage'
+    ];
+
+    /**
+     * @var DataObject|RecursivePublishable
+     */
+    protected $record;
+
+    /**
+     * @param string $name
+     * @param DataObject|RecursivePublishable $record
+     */
+    public function __construct($name, $record)
+    {
+        $this->record = $record;
+
+        parent::__construct($name, null);
+    }
+
+    /**
+     * @return array
+     */
+    public function getSchemaDataDefaults()
+    {
+        $defaults = parent::getSchemaDataDefaults();
+        $record = $this->getRecord();
+        $usageLink = $this->Link('usage');
+
+        $defaults['data'] = array_merge($defaults['data'], [
+            'recordId' => $record->ID,
+            'recordClass' => get_class($record),
+            'readUsageEndpoint' => [
+                'url' => $usageLink,
+                'method' => 'get',
+            ]
+        ]);
+
+        return $defaults;
+    }
+
+    /**
+     * @param HTTPRequest $request
+     * @return HTTPResponse $response
+     */
+    public function usage(HTTPRequest $request)
+    {
+        $record = $this->getRecord();
+        $usage = $record->findOwners();
+
+        $this->extend('updateUsage', $usage, $record);
+
+        $usageData = [];
+        /**
+         * @var string $userId
+         * @var DataObject $user
+         */
+        foreach ($usage as $userId => $user) {
+            $state = $this->getState($user);
+            $link = $user->hasMethod('CMSEditLink') ? $user->CMSEditLink() : null;
+
+            $usageData[] = [
+                'id' => $userId,
+                'title' => $user->getTitle(),
+                'type' => $user->singular_name(),
+                'state' => $state,
+                'link' => $link,
+            ];
+        }
+
+        $data = [
+            'usage' => $usageData,
+        ];
+
+        $response = new HTTPResponse(json_encode($data));
+        return $response
+            ->addHeader('Content-Type', 'application/json');
+    }
+
+    /**
+     * @param $record
+     * @return null|string
+     */
+    protected function getState($record)
+    {
+        if (!$record->hasExtension(Versioned::class)) {
+            return null;
+        }
+
+        if ($record->isOnDraftOnly()) {
+            return _t(__CLASS__ . '.ADDEDTODRAFT', 'Draft');
+        }
+
+        if ($record->isModifiedOnDraft()) {
+            return _t(__CLASS__ . '.MODIFIEDONDRAFTSHORT', 'Modified');
+        }
+
+        return null;
+    }
+
+    /**
+     * @return DataObject|RecursivePublishable
+     */
+    public function getRecord()
+    {
+        return $this->record;
```

Comment:
I think we need a setter as well.

Reply:
See other note; If $this->record isn't set, you should lazy-infer this from the form record.